rules/windows/other/win_wmi_persistence.yml

```diff
@@ -1,33 +1,62 @@
+action: global
 title: WMI Persistence
 id: 0b7889b4-5577-4521-a60a-3376ee7f9f7b
 status: experimental
-description: Detects suspicious WMI event filter and command line event consumer based on event id 5861 and 5859 (Windows 10, 2012 and higher)
-author: Florian Roth
+description: Detects suspicious WMI event filter and command line event consumer based on WMI and Sysmon logs.
+author: Florian Roth, Gleb Sukhodolskiy, Timur Zinniatullin oscd.community
 date: 2017/08/22
-modified: 2020/08/23
+modified: 2020/1013
 references:
     - https://twitter.com/mattifestation/status/899646620148539397
     - https://www.eideon.com/2018-03-02-THL03-WMIBackdoors/
+    - https://attack.mitre.org/techniques/T1546/003/
 tags:
     - attack.persistence
     - attack.privilege_escalation
     - attack.t1084           # an old one
     - attack.t1546.003
+falsepositives:
+    - Unknown (data set is too small; further testing needed)
+level: medium
+---
 logsource:
     product: windows
-    service: wmi
+    service: wmi #native windows detection
+    definition: 'WMI Namespaces Auditing and SACL should be configured, EventID 5861 and 5859 detection requires Windows 10, 2012 and higher'
 detection:
-    selection:
+    wmi_activity_5861:
         EventID: 5861
     keywords:
         Message:
             - '*ActiveScriptEventConsumer*'
             - '*CommandLineEventConsumer*'
             - '*CommandLineTemplate*'
         # - 'Binding EventFilter'  # too many false positive with HP Health Driver
-    selection2:
+    wmi_activity_5859:
         EventID: 5859
-    condition: selection and 1 of keywords or selection2
-falsepositives:
-    - Unknown (data set is too small; further testing needed)
-level: medium
+    network_logon:
+        EventID: 4624
+        LogonType: 3
+    privileges_assigned:
+        EventID: 4672
+    wmi_subscription:
+        EventID: 4662
+        ObjectType: 'WMI Namespace'
+        ObjectName: '*subscription*'
+    condition: (wmi_activity_5861 and 1 of keywords) OR (wmi_activity_5859) OR (network_logon and privileges_assigned and wmi_subscription)
+---
+logsource:
+    product: windows
+    service: sysmon #sysmon detection
+detection: 
```

Comment:
Hey @GlebSukhodolskiy, sysmon Events are already covered in https://github.com/Neo23x0/sigma/blob/master/rules/windows/sysmon/sysmon_wmi_event_subscription.yml

Reply:
Hey! I should delete that part from the rule? Or we can keep my variant and deprecate an old one?

Reply:
I think it makes sense to remove, what do you think @aw350m33d?

Reply:
Although it would be reasonable to collect logic in a single place to identify the same action, maintainers don't like to delete existing rules. Moreover, there is a separate folder for rules based on sysmon events. But I think that this issue can be discussed with maintainers in order to develop general recommendations for such cases. For now, I propose to remove the duplication of the existing rule from the current rule.

Jonhnathan (@w0rk3r), thank you very much for participating in the discussions. 👍🏻 And for a hint about the existence of a rule on sнsmon events. It is very important for us that the participants take part and communicate in the comments, looked at each other's rules, and expressed your opinion.


Gleb there is also a small problem with the log sources in the first part of the rule. The fact is that service is indicated there `service: wmi`. This means that events `4624`, `4672` and `4662` will not be there, since they refer to the security log (`service: security`). Therefore, the first part should be split into two elements of the rule collection. 

I also thank you a lot for using meaningful search ID names! 🥇 I am absolutely convinced that it is bad practice to use something like selection_1, selection2, or something similar for **complex** constructsions. I believe the search ID names should be self-explanatory. Then the expression in the `condition` attribute can be read as the text of the natural language. I suggest the following option:
```
---
logsource:
    product: windows
    service: wmi #native windows detection
    definition: 'WMI Namespaces Auditing and SACL should be configured, EventID 5861 and 5859 detection requires Windows 10, 2012 and higher'
detection:
    wmi_filter_to_consumer_binding:
        EventID: 5861
    keywords:
        Message:
            - '*ActiveScriptEventConsumer*'
            - '*CommandLineEventConsumer*'
            - '*CommandLineTemplate*'
        # - 'Binding EventFilter'  # too many false positive with HP Health Driver
    wmi_filter_registration:
        EventID: 5859
    condition: (wmi_filter_to_consumer_binding and 1 of keywords) OR (wmi_filter_registration)
---
logsource:
    product: windows
    service: security
detection:
    network_logon:
        EventID: 4624
        LogonType: 3
    privileges_assigned:
        EventID: 4672
    wmi_subscription:
        EventID: 4662
        ObjectType: 'WMI Namespace'
        ObjectName: '*subscription*'
    condition: network_logon and privileges_assigned and wmi_subscription
---
```